Orange/widgets/utils/itemmodels.py

```diff
@@ -145,6 +154,7 @@ def sort(self, column, order=Qt.AscendingOrder):
                          key=lambda i: self._table[i][column],
                          reverse=order != Qt.AscendingOrder)
         self._table[:] = [self._table[i] for i in indices]
+        self._roleData[:] = [self._roleData[i] for i in indices]
```

Comment:
As I said, I may have missed something. I overlooked that the constructor calls `wrap`, and that calling `wrap` is also the only (legal) way of assigning to `_table`.